fail build

~~~
-- The C compiler identification is GNU 11.1.0
-- The CXX compiler identification is GNU 11.1.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
--  Found clang-format version <12.0.1>.
--  Using .clang-format version 10.
CMAKE_CXX_STANDARD = <11>
-- Configuring done
-- Generating done
-- Build files have been written to: /tmp/makepkg/sl1-vulkan-git/src/Vulkan-Hpp/build
[ 33%] Building CXX object CMakeFiles/VulkanHppGenerator.dir/VulkanHppGenerator.cpp.o
/tmp/makepkg/sl1-vulkan-git/src/Vulkan-Hpp/VulkanHppGenerator.cpp: In member function 'std::string VulkanHppGenerator::generateRAIIHandleConstructorArguments(const string&, const std::vector<VulkanHppGenerator::ParamData>&, bool, bool) const':
/tmp/makepkg/sl1-vulkan-git/src/Vulkan-Hpp/VulkanHppGenerator.cpp:8031:14: error: variable 'typeIt' set but not used [-Werror=unused-but-set-variable]
 8031 |         auto typeIt = std::find_if(
      |              ^~~~~~
/tmp/makepkg/sl1-vulkan-git/src/Vulkan-Hpp/VulkanHppGenerator.cpp: In member function 'std::string VulkanHppGenerator::generateRAIIHandleConstructorVector(const std::pair<std::__cxx11::basic_string<char>, VulkanHppGenerator::HandleData>&, std::map<std::__cxx11::basic_string<char>, VulkanHppGenerator::CommandData>::const_iterator, std::vector<VulkanHppGenerator::ParamData>::const_iterator, const string&, const string&) const':
/tmp/makepkg/sl1-vulkan-git/src/Vulkan-Hpp/VulkanHppGenerator.cpp:8484:10: error: variable 'structureIt' set but not used [-Werror=unused-but-set-variable]
 8484 |     auto structureIt = m_structures.find( lenIt->type.type );
      |          ^~~~~~~~~~~
cc1plus: all warnings being treated as errors
make[2]: *** [CMakeFiles/VulkanHppGenerator.dir/build.make:76: CMakeFiles/VulkanHppGenerator.dir/VulkanHppGenerator.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:83: CMakeFiles/VulkanHppGenerator.dir/all] Error 2
make: *** [Makefile:91: all] Error 2
~~~

gcc 11.1.0
clang-format 12.0.1

--- a/VulkanHppGenerator.cpp
+++ b/VulkanHppGenerator.cpp
@@ -8097,11 +8097,13 @@ std::string VulkanHppGenerator::generateRAIIHandleConstructorArguments( std::str
            ( specialPointerTypes.find( param.type.type ) == specialPointerTypes.end() ) )
       {
         // this is supposed to be the returned size on an enumeration function!
+#if !defined(NDEBUG)
         assert( param.type.type == "uint32_t" );
         auto typeIt = std::find_if(
           params.begin(), params.end(), [&handleType]( ParamData const & pd ) { return pd.type.type == handleType; } );
         assert( typeIt != params.end() );
         assert( typeIt->len == param.name );
+#endif
         continue;
       }
       else if ( std::find_if( params.begin(),
@@ -8550,6 +8552,7 @@ std::string VulkanHppGenerator::generateRAIIHandleConstructorVector(
     lenIt = std::find_if( constructorIt->second.params.begin(),
                           constructorIt->second.params.end(),
                           [&lenParts]( ParamData const & pd ) { return pd.name == lenParts[0]; } );
+#if !defined(NDEBUG)
     assert( lenIt != constructorIt->second.params.end() );
     auto structureIt = m_structures.find( lenIt->type.type );
     assert( structureIt != m_structures.end() );
@@ -8558,6 +8561,7 @@ std::string VulkanHppGenerator::generateRAIIHandleConstructorVector(
                           [&lenParts]( MemberData const & md )
                           { return md.name == lenParts[1]; } ) != structureIt->second.members.end() );
     assert( constructorIt->second.successCodes.size() == 1 );
+#endif
     vectorSize = startLowerCase( stripPrefix( lenParts[0], "p" ) ) + "." + lenParts[1];
   }
   else

